src/formatting-model-api.d.ts

```diff
@@ -0,0 +1,582 @@
+/**
+ * Formatting Pane Model Interfaces
+ */
+
+declare module powerbi {
+    export namespace visuals {
+        export type FormattingModelWarnings = { [cardUid: string]: IVisualWarning };
+
+        export interface FormattingModel {
+            cards: (FormattingCard | FormattingCardPlaceholder)[];
+            modelUnavailableImageName?: string;
+            modelUnavailableMessageKey?: string;
+            modelUnavailableLearnMoreLink?: string;
+            // this is used for displaying messages in formatting pane, with an action to usually open some dialog.
+            formattingModelMessage?: FormattingModelMessage;
+            warnings?: FormattingModelWarnings;
+        }
+
+        export interface FormattingModelMessage {
+            messageKey: string;
+            action?: FormattingModelMessageAction;
+        }
+
+        export interface FormattingModelMessageAction {
+            key: string;
+            fn: () => void;
```

Comment:
what does fn() do? does the CV pass the code into the host?

Reply:
I dont know how it exactly works, But I see that yes it run some code in our side in some html file in front end.
Should I delete it ?

Reply:
if this is code that the visual provides then we should remove it.